### PR TITLE
a trivial fix for distinct and groupby without aggregates before DAG

### PR DESCRIPTION
**Pull request by ilovesoup**
For now distinct and group by without aggregates will fail since Coprocessor needs a aggregate function to work when grouping.
Add fake first() in such cases and skip it in schema infer.

Refer to https://github.com/pingcap/tispark/pull/90/files

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/156)

### CODE REVIEW

File: src/main/java/com/pingcap/tikv/meta/TiSelectRequest.java [hunk 1]
```diff
@@ -88,6 +91,14 @@ public SelectRequest buildScan(boolean idxScan) {
     }
   }
 
+  public void skipAggregatesAt(int ordinal) {
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/meta/TiSelectRequest.java:94]: didn't see any usage of this function. Why add this? I got the method's meaning but don't know what does `ordinal` mean.

File: src/main/java/com/pingcap/tikv/operation/SchemaInfer.java [hunk 2]
```diff
@@ -75,8 +75,13 @@ private void buildTransform(TiSelectRequest tiSelectRequest) {
 
     // append aggregates if present
     if (!tiSelectRequest.getAggregatePairs().isEmpty()) {
-      for (Pair<TiExpr, DataType> pair : tiSelectRequest.getAggregatePairs()) {
-        rowTrans.addProjection(new Cast(pair.second));
+      for (int i = 0; i < tiSelectRequest.getAggregatePairs().size(); i++) {
+        if (!tiSelectRequest.isAggregatesSkipped(i)) {
+          Pair<TiExpr, DataType> pair = tiSelectRequest.getAggregatePairs().get(i);
```
Comment (zhexuany) [inline src/main/java/com/pingcap/tikv/operation/SchemaInfer.java:80]: Should we check `getAggregatePairs()` is null or not?

File: src/main/java/com/pingcap/tikv/meta/TiSelectRequest.java (hunk 1 above)
Comment (ilovesoup) [reply src/main/java/com/pingcap/tikv/meta/TiSelectRequest.java:94]: Used in TiSpark. Ordinal means position: first, second and etc.

File: src/main/java/com/pingcap/tikv/operation/SchemaInfer.java (hunk 2 above)
Comment (ilovesoup) [reply src/main/java/com/pingcap/tikv/operation/SchemaInfer.java:80]: No
private final List<Pair<TiExpr, DataType>> aggregates = new ArrayList<>();

**Review by birdstorm — APPROVED** [review]
LGTM

**Pull request merged**